src/Features/Core/Portable/CodeAnalysisSuggestions/ICodeAnalysisSuggestionsConfigService.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CodeAnalysisSuggestions;
+
+/// <summary>
+/// Service to compute project-wide code analysis diagnostics for first party analyzers/fixes that are relevant
+/// for the user and/or the code base to improve the code quality and/or code style of the code base.
+/// </summary>
+internal interface ICodeAnalysisSuggestionsConfigService : IWorkspaceService
+{
+    /// <summary>
+    /// Compute project-wide code analysis diagnostics for first party analyzers/fixes that are relevant
+    /// for the user and/or the code base to improve the code quality and/or code style of the code base.
+    /// </summary>
+    /// <returns>
+    /// Returns an array of tuples '(Category, DiagnosticsById)', where 'Category' is the diagnostic category
+    /// of the corresponding diagnostics in the 'DiagnosticsById' map. 'DiagnosticsById' is a dictionary from
+    /// diagnostic ID to array of diagnostics with that ID that are reported in the current project.
+    /// </returns>
+    public Task<ImmutableArray<(string Category, ImmutableDictionary<string, ImmutableArray<DiagnosticData>> DiagnosticsById)>> TryGetCodeAnalysisSuggestionsConfigDataAsync(Project project, CancellationToken cancellationToken);
```

Comment:
do we need DiagnosticsById in the return value as well?  if it's alreayd stored in that fashion, it's fine. but if internally it isn't and is just stored as an array, might as well return that.  the caller can group by diagnosticData-id if they want.

Reply:
Thanks, done.